scipy/stats/_distn_infrastructure.py

```diff
@@ -1240,8 +1240,13 @@ def moment(self, n, *args, **kwds):
 
         """
         args, loc, scale = self._parse_args(*args, **kwds)
-        if not (self._argcheck(*args) and (scale > 0)):
-            return nan
+        arrs = tuple(map(np.asarray, [*args, loc, scale]))
+        arrs = np.broadcast_arrays(*arrs)
+        args = arrs[:-2]
+        loc, scale = arrs[-2], arrs[-1]
+        cond0 = (self._argcheck(*args) and scale > 0)
```

Comment:
```suggestion
        cond0 = (self._argcheck(*args) & scale > 0)
```
Currently isn't working when there are shapes:
```
import numpy as np
from scipy.stats import chi as dist
m = dist.moment(n=3, df = 1, loc=[1, 2, 3], scale=1)
```

Reply:
Otherwise, this stuff up top looks good to me.

Reply:
I guess this is fixed now.